src/main/java/dev/coms4156/project/kebabcase/controller/BuildingController.java

```diff
@@ -519,18 +519,26 @@ public ResponseEntity<?> addExistingBuildingToUser(@PathVariable int userId,
    * Retrieves a list of all buildings from the repository,
    * returns all buildings in the repository as a list. 
    *
+   * @param  address an optional request parameter to select buildings containing
+   *         the specified address.
    * @return ResponseEntity containing the list of buildings as a JSON response.
    *         Returns a 200 OK status if buildings are found, or 204 No Content if no 
    *         buildings exist in the repository.
    */
   @GetMapping("/buildings")
-  public ResponseEntity<List<BuildingEntity>> getBuildings() {
+  public ResponseEntity<List<BuildingEntity>> getBuildings(
+           @RequestParam(required = false) String address) {
     List<BuildingEntity> buildings = buildingRepository.findAll();
 
     if (buildings.isEmpty()) {
       return ResponseEntity.status(HttpStatus.NO_CONTENT).body(null);
     }
 
+    if (address != null && !address.isEmpty()) {
```

Comment:
So, we really don't want to run buildingRepository.findAll() if we don't have to.  (And we don't have to when address is not empty. 

So we need to reorder the logic here.